### PR TITLE
Prefer local Prettier when formatting files

### LINKED ISSUE #616: prettier old version
## Affected Packages

apply-release-plan

## Problem

It's relying on an old version of prettier: https://github.com/atlassian/changesets/blob/main/packages/apply-release-plan/package.json#L20

## Proposed solution

Is it necessary to ship prettier as a dependency of this package? Could it be moved to devDependencies? If not, upgrade prettier to v2.

### PR DESCRIPTION
**Pull request by Andarist**
closes #616

it's the best to just take a look at the first commit (the one without formatting changes): https://github.com/changesets/changesets/pull/905/commits/85fa88920e13a188e478824eebaf4c81e5b0d90a

### CODE REVIEW

File: packages/apply-release-plan/src/index.ts [hunk 1]
```diff
@@ -19,14 +19,25 @@ import prettier from "prettier";
 import versionPackage from "./version-package";
 import getChangelogEntry from "./get-changelog-entry";
 
+function getPrettierInstance(): typeof prettier {
+  try {
+    return require("prettier");
```
Comment (mitchellhamilton) [inline packages/apply-release-plan/src/index.ts:24]: Could we `createRequire` and resolve it from the repo root? Without that, this doesn't change anything, right?

Comment (mitchellhamilton) [reply packages/apply-release-plan/src/index.ts:24]: Looks like `require.resolve("prettier", { paths: [projectDir] })` works on more versions so let's probably use that

Comment (Andarist) [reply packages/apply-release-plan/src/index.ts]: Ah, ye - this is totally what I had in mind when I started working on this and just forgot it in the rush 🤣

File: packages/write/src/index.ts [hunk 2]
```diff
@@ -4,6 +4,17 @@ import prettier from "prettier";
 import humanId from "human-id";
 import { Changeset } from "@changesets/types";
 
+function getPrettierInstance(): typeof prettier {
+  try {
+    return require("prettier");
```
Comment (mitchellhamilton) [inline packages/write/src/index.ts:9]: This also needs to use `require.resolve`

Comment (Andarist) [reply packages/write/src/index.ts]: fixed

**Review by mitchellhamilton — APPROVED** [review]

**Review by Samira900i8uu — APPROVED** [review]

**Pull request merged**